File: Data Structures & Algorithms/is-anagram/submission-0.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        if len(s) != len(t):
            return False
        sc = {}
        tc = {}    

        for i in s :
            if i in sc:
                sc[i] += 1
            else :
                sc[i] = 1

        for i in t :
            if i in tc :
                tc[i] += 1
            else :
                tc[i] = 1

        for i in sc :
            if i not in tc or sc[i] != tc[i]:
                return False

        return True



        

File: Data Structures & Algorithms/single-number/submission-0.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {number}
     */
    singleNumber(nums) {
        const obj = {}
        for (let i =0;i<nums.length;i++){
            obj[nums[i]] = (obj[nums[i]] || 0) + 1
        }
        console.log(obj)
        for (let i in obj){
            if (obj[i] == 1) {
                return i
            }
        }
    }
}
